name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/payments/PaymentsPerMonthChartBuilder.java

```diff
@@ -144,25 +146,45 @@ public void configure(Chart chart, Consumer<TabularDataSource> selectionListener
                         new DividendPerMonthChartToolTip(chart, selectionListener));
     }
 
+    /**
+     * Creates bar series for each year in the provided chart, with bars
+     * representing monthly payment data. Uses a TreeMap to automatically sort
+     * the years in ascending order.
+     */
     @Override
     public void createSeries(Chart chart, PaymentsViewModel model)
     {
         chart.setData(PaymentsViewModel.class.getSimpleName(), model);
 
+        // Use a TreeMap to automatically sort the years
+        Map<Integer, IBarSeries> yearToSeries = new TreeMap<>();
```

Comment:
I do not think the month is ever in the wrong order, because it is never using a stacked series.
Plus: the order is determined by the #createSeries call. The TreeMap does not make a difference here.